src/uct/base/uct_iface.c

```diff
@@ -273,6 +261,19 @@ int uct_iface_is_reachable_v2(const uct_iface_h tl_iface,
                               const uct_iface_is_reachable_params_t *params)
 {
     const uct_base_iface_t *iface = ucs_derived_of(tl_iface, uct_base_iface_t);
+
+    if (!ucs_test_all_flags(params->field_mask,
+                            UCT_IFACE_IS_REACHABLE_FIELD_IFACE_ADDR |
+                            UCT_IFACE_IS_REACHABLE_FIELD_DEVICE_ADDR)) {
```

Comment:
maybe redefine API so that `iface` and `device` addresses are provided as separate parameters?
they are mandatory anyway..

Reply:
should it be done it this PR or a separated one?

Reply:
question is if we think they will always be mandatory in the future.

Reply:
IMO current impl is good, I won't redefine.